service/service.go

```diff
@@ -0,0 +1,58 @@
+package service
+
+import (
+	"encoding/json"
+	"github.com/Jeffail/gabs"
+	"github.com/blockassets/cgminer_client"
+	"io"
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"time"
+)
+
+//TODO: refactor this file to separate out the service level commands into their own files
+type Commands interface {
+	CgmQuit() error
+	Reboot()
+	UpdatePools(poolsAsJson io.ReadCloser, configFilePath string) error
+}
+type Command struct {
+}
+
+func (*Command) CgmQuit() error {
+	clnt := cgminer_client.New("localhost", 4028, 5*time.Second)
+	return clnt.Quit()
+}
+
+func (*Command) Reboot() {
+	time.Sleep(5 * time.Second)
+	log.Printf("Reboot Requested")
+	exec.Command("/sbin/reboot", "-f").Run()
+}
+
+type PoolAddresses struct {
+	Pool1 string `json:"pool1"`
+	Pool2 string `json:"pool2"`
+	Pool3 string `json:"pool3"`
+}
+
+func (*Command) UpdatePools(poolsAsJson io.ReadCloser, configFilePath string) error {
+	var pools PoolAddresses
+	err := json.NewDecoder(poolsAsJson).Decode(&pools)
+	if err == nil {
+		// use the gabs json helper..
+		cont := gabs.New()
+		cont, err = gabs.ParseJSONFile(configFilePath)
+		if err == nil {
+			cont.SetP(pools.Pool1, "pool1")
+			cont.SetP(pools.Pool2, "pool2")
+			cont.SetP(pools.Pool3, "pool3")
+			err = ioutil.WriteFile(configFilePath, cont.Bytes(), 0644)
+		}
+	}
+	if err != nil {
```

Comment:
In light of the lack of exceptions in this language, I prefer the check for error to be closer to the execution.

Reply:
This is because the check for error is done immediately, except the last err.
But I want to do log the error in every case...

Its seems ugly to repeat the log everytime.

We could use defer, panic and recover to do the same thing... (ie manage using go equivelent of exceptions.... but I think we should have  alonger discussion over a beer when you are back on that